File: app/src/main/java/vasuki/istanpdf/domain/MergePdf.java
package vasuki.istanpdf.domain;

import android.net.Uri;

import java.util.List;

import vasuki.istanpdf.data.PdfEngine;

public final class MergePdf {
    private final PdfEngine pdfEngine;

    public MergePdf(PdfEngine pdfEngine) {
        this.pdfEngine = pdfEngine;
    }

    public void execute(List<Uri> sources, List<Integer> rotations, Uri destination) throws Exception {
        if (sources == null || sources.size() < 2) {
            throw new IllegalArgumentException("At least 2 PDFs are required to merge");
        }
        pdfEngine.merge(sources, rotations, destination);
    }
}

File: app/src/main/kotlin/vasuki/istanpdf/domain/MergePdf.kt
package vasuki.istanpdf.domain

import android.net.Uri
import vasuki.istanpdf.data.PdfEngine

class MergePdf(private val pdfEngine: PdfEngine) {
    @Throws(Exception::class)
    fun execute(sources: List<Uri>, rotations: List<Int>, destination: Uri) {
        require(sources.size >= 2) { "At least 2 PDFs are required to merge" }
        pdfEngine.merge(sources, rotations, destination)
    }
}
